app/renderer/ui/components/list/index.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react"
+import { List } from "antd"
+
+import { ButtonOption } from "app/renderer/ui/components/button"
+
+const styles = require("./style.scss")
+
+export interface Iprops {
+  grid?: {} | undefined
+  classNameList?: string
+  classNameItem?: string
+  dataSource: Array<(string | { text: any; onClick: any } | any)>
+ // dataSource: (string | { text: any; onClick: any })[]
+  borderer?: boolean
+  renderItem?: any
+}
+
+/**
+ * List UI component
+ *
+ * @export
+ * @class List
+ * @extends {React.Component<Iprops>}
+ */
+
+export default class DefaultList extends React.Component<Iprops> {
+  // tslint:disable-next-line: completed-docs
+  public render() {
+    const ListItem = this.props.renderItem
+    const isClickable = (item: any) => item.onClick && item.text
+    const renderItem = (item: any) => {
+      return isClickable(item)
+        ? (
```

Comment:
This ternary conditional is overly complex. We can extract its two expressions into separate functions parameterized over `props` and `item`.